### PR TITLE
Particle Accelerator Console TGUI update

### PR DESCRIPTION
**Pull request by Migratingcocofruit**
## What Does This PR Do
Adds a layout of the particle accelerator to the console with tooltips for each part that explain what is wrong with it.
There are also coloured borders:
Green means it's good
Orange means it's oriented correctly, but needs to be wrenched down/wired/etc
Red means it's either not oriented correctly or is missing or at the wrong position.

## Why It's Good For The Game
The Particle accelerator can be a bit of a pain to set up, especially for players who don't use it often. For some parts it's not entirely clear which part is the front, and besides that, having an in-game indication to what's wrong is good.

## Images of changes
image

## Testing
Made the particle accelerator wrong and right in various ways and observed the expected images and tooltips showing up on the console

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
add: Added a Particle Accelerator layout and indication to its console
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 1]
```diff
@@ -42,6 +80,140 @@ export const ParticleAccelerator = (props, context) => {
             </LabeledList.Item>
           </LabeledList>
         </Section>
+        <Section
+          title={
+            orientation ? 'EM Acceleration Chamber Orientation: ' + orientation : 'No EM Acceleration Chamber Detected'
+          }
+        />
```
[the remark this answers is not in the document]
Comment (Migratingcocofruit) [reply tgui/packages/tgui/interfaces/ParticleAccelerator.js:87]: Probably better. I'll give it a shot.

**Review by Burzah — COMMENTED** [review]
I would like to see the cardinal directions displayed in the second section title of the UI use capitalization.

**Comment by Migratingcocofruit** [comment]
> I would like to see the cardinal directions displayed in the second section title of the UI use capitalization.

done

**Review by Burzah — APPROVED** [review]

**Comment by Burzah** [comment]
The UI looks good for this, but I would like another reviewer to go over the rest of this.

**Review by Burzah — CHANGES_REQUESTED** [review]
Removing approval for the time being. After testing this, there seems to be some issues with orientation detection. You can get it to work, but it takes a bunch of connecting and reconnecting to get the orientation to be recognized properly by the console. Also there looks to be an issue with the Right Containment Grid when it is facing towards the East, and there is improper detection with the orientation of both Left and Right containment grids. Please look for any issues and retest this.

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 2]
```diff
@@ -42,7 +81,158 @@ export const ParticleAccelerator = (props, context) => {
...
+
+const LayoutHorizontal = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { assembled, power, strength, max_strength, icon, layout_1, layout_2, layout_3, orientation } = data;
+  return (
+    <Table>
+      <TableRow width="40px">
+        {(orientation === 'east' ? layout_1 : layout_3).slice().map((item) => (
+          <Table.Cell key={item.name}>
+            <Tooltip content={`${item.name} Status:${item.status}, Direction:${dir2text(item.dir)}`}>
```
Comment (Burzah) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:116]:
```suggestion
            <Tooltip content={`${item.name} Status: ${item.status}, Direction: ${dir2text(item.dir)}`}>
```

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 3]
```diff
@@ -42,7 +81,158 @@ export const ParticleAccelerator = (props, context) => {
...
+                }}
+              />
+            </Tooltip>
+          </Table.Cell>
+        ))}
+      </TableRow>
+      <TableRow width="40px">
+        {layout_2.slice().map((item) => (
+          <Table.Cell key={item.name}>
+            <Tooltip content={`${item.name} Status:${item.status}, Direction:${dir2text(item.dir)}`}>
```
Comment (Burzah) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:135]:
```suggestion
            <Tooltip content={`${item.name} Status: ${item.status}, Direction: ${dir2text(item.dir)}`}>
```

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 4]
```diff
@@ -42,7 +81,158 @@ export const ParticleAccelerator = (props, context) => {
...
+                }}
+              />
+            </Tooltip>
+          </Table.Cell>
+        ))}
+      </TableRow>
+      <TableRow width="40px">
+        {(orientation === 'east' ? layout_3 : layout_1).slice().map((item) => (
+          <Table.Cell key={item.name}>
+            <Tooltip content={`${item.name} Status:${item.status}, Direction:${dir2text(item.dir)}`}>
```
Comment (Burzah) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:154]:
```suggestion
            <Tooltip content={`${item.name} Status: ${item.status}, Direction: ${dir2text(item.dir)}`}>
```

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 5]
```diff
@@ -42,7 +81,158 @@ export const ParticleAccelerator = (props, context) => {
...
+
+const LayoutVertical = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { assembled, power, strength, max_strength, icon, layout_1, layout_2, layout_3, orientation } = data;
+  return (
+    <Grid>
+      <GridColumn width="40px">
+        {(orientation === 'north' ? layout_1 : layout_3).slice().map((item) => (
+          <Stack.Item grow key={item.name}>
+            <Tooltip content={`${item.name} Status:${item.status}, Direction:${dir2text(item.dir)}`}>
```
Comment (Burzah) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:182]:
```suggestion
            <Tooltip content={`${item.name} Status: ${item.status}, Direction: ${dir2text(item.dir)}`}>
```

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 6]
```diff
@@ -42,7 +81,158 @@ export const ParticleAccelerator = (props, context) => {
...
+                }}
+              />
+            </Tooltip>
+          </Stack.Item>
+        ))}
+      </GridColumn>
+      <GridColumn>
+        {layout_2.slice().map((item) => (
+          <Stack.Item grow key={item.name}>
+            <Tooltip content={`${item.name} Status:${item.status}, Direction:${dir2text(item.dir)}`}>
```
Comment (Burzah) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:201]:
```suggestion
            <Tooltip content={`${item.name} Status: ${item.status}, Direction: ${dir2text(item.dir)}`}>
```

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js [hunk 7]
```diff
@@ -42,7 +81,158 @@ export const ParticleAccelerator = (props, context) => {
...
+                }}
+              />
+            </Tooltip>
+          </Stack.Item>
+        ))}
+      </GridColumn>
+      <GridColumn width="40px">
+        {(orientation === 'north' ? layout_3 : layout_1).slice().map((item) => (
+          <Stack.Item grow key={item.name} tooltip={item.status}>
+            <Tooltip content={`${item.name} Status:${item.status}, Direction:${dir2text(item.dir)}`}>
```
Comment (Burzah) [inline tgui/packages/tgui/interfaces/ParticleAccelerator.js:220]:
```suggestion
            <Tooltip content={`${item.name} Status: ${item.status}, Direction: ${dir2text(item.dir)}`}>
```

**Review by Burzah — APPROVED** [review]
Ran another test of this today, looks good. I have just minor suggestions here for the tooltip stuff. I am sure this could be made more DRY, but it is working as we discussed so I am all for moving forward with it.

File: tgui/packages/tgui/interfaces/ParticleAccelerator.js (hunk 2 above)
Comment (Migratingcocofruit) [reply tgui/packages/tgui/interfaces/ParticleAccelerator.js:116]: The lack of spaces there is to keep category and value on the same line. I tried separating the lines manually with \n so I could put spaces there but it didn't work.

**Pull request merged**